opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -82,48 +82,59 @@
 ParentSpan = typing.Optional[typing.Union["Span", "SpanContext"]]
 
 
-class Link:
-    """A link to a `Span`."""
-
-    def __init__(
-        self, context: "SpanContext", attributes: types.Attributes = None
-    ) -> None:
+class LinkBase(abc.ABC):
+    def __init__(self, context: "SpanContext") -> None:
         self._context = context
-        if attributes is None:
-            self._attributes = {}  # type: types.Attributes
-        else:
-            self._attributes = attributes
 
     @property
     def context(self) -> "SpanContext":
         return self._context
 
     @property
```

Comment:
Having `@property` along with an `@abstractmethod` does not do much:

```python
from abc import ABC, abstractmethod


class Parent(ABC):

    @property
    @abstractmethod
    def attribute(self):
        pass


class Child0(Parent):
    def attribute(self):
        return "attribute"


class Child1(Parent):
    @property
    def attribute(self):
        return "attribute"


print(type(Child0().attribute))
print(type(Child1().attribute))
```
```
<bound method Child0.attribute of <__main__.Child0 object at 0x7fa980d64890>>
attribute
```

Reply:
Could you please explain what's exactly wrong there?
I think I'm doing it according to the documentation https://docs.python.org/3/library/abc.html#abc.abstractproperty.